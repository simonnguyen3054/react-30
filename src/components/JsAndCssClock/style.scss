.clockBackground {
  background: url("../../assets/images/livingroom.jpg");
  background-size: cover;
  min-height: 100vh;
  width: 100%;

  .clock {
    width: 26rem;
    height: 26rem;
    background-color: #997272;
    clip-path: circle(50% at 50% 50%);
    display: flex;
    justify-content: center;
    align-items: center;
    box-shadow: 10px -10px rgba(0, 0, 0, 0.6);
    position: relative;
    left: 60rem;
    top: 5rem;

    &-inner {
      width: 25rem;
      height: 25rem;
      // border: 8px solid #857d8e;
      // border-radius: 50%;
      // margin: 50px auto;
      // position: relative;
      background: url("../../assets/images/cart.jpg") no-repeat;
      background-size: cover;
      clip-path: circle(50% at 50% 50%);
      background-position: 50% 95%;
    }

    &-face {
      position: relative;
      width: 100%;
      height: 100%;
      transform: translateY(-3px); /* account for the height of the clock hands */
    }

    .hand {
      width:50%;
      height:6px;
      background: #cdd2d6;
      position: absolute;
      top:50%;
      border-radius: 5px;
    }
  }
}
