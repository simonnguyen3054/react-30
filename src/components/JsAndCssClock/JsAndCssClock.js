import React from "react";
import "./style.scss";

const JsAndCssClock = () => {
  return (
    <div className="clockBackground">
      <div className="clock">
        <div className="clock-inner">
          <div className="clock-face">
            <div className="hand hour-hand"></div>
            <div className="hand min-hand"></div>
            <div className="hand second-hand"></div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default JsAndCssClock;
