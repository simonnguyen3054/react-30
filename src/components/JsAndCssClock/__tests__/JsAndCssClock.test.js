import React from "react";
import { mount } from "enzyme";
import JsAndCssClock from "../JsAndCssClock";

let wrapper = null;
beforeEach(() => {
  wrapper = mount(<JsAndCssClock />);
});

afterEach(() => {
  wrapper.unmount();
});

describe("JsAndCssClock", () => {
  it("should render the clock on the wall", () => {
    expect(wrapper).toMatchSnapshot();
  });
});