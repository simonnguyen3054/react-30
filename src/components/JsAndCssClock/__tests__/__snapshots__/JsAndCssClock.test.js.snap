// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsAndCssClock should render the clock on the wall 1`] = `
<JsAndCssClock>
  <div
    className="clockBackground"
  >
    <div
      className="clock"
    >
      <div
        className="clock-inner"
      >
        <div
          className="clock-face"
        >
          <div
            className="hand hour-hand"
            style={
              Object {
                "transform": "rotate(90deg)",
              }
            }
          />
          <div
            className="hand min-hand"
            style={
              Object {
                "transform": "rotate(90deg)",
              }
            }
          />
          <div
            className="hand sec-hand"
            style={
              Object {
                "transform": "rotate(90deg)",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</JsAndCssClock>
`;
